src/app/info/AddWikiItemButton.tsx

```diff
@@ -0,0 +1,52 @@
+"use client";
+
+import { DbWikiRow } from "@/databaseUtils";
+import { ButtonMargin } from "@/app/info/StyleComponents";
+import { Button } from "@mui/material";
+import supabase from "@/supabaseClient";
+import { logErrorReturnLogId } from "@/logger/logger";
+import { PostgrestError } from "@supabase/supabase-js";
+
+interface WikiRowQuerySuccessType {
+    data: DbWikiRow;
+    error: null;
+}
+
+interface WikiRowQueryFailureType {
+    data: null;
+    error: PostgrestError;
+}
+export type WikiRowQueryType = WikiRowQuerySuccessType | WikiRowQueryFailureType;
+
+interface AddWikiItemButtonProps {
+    doesEmptyRowExist: boolean;
+    appendNewRow: (newRow: DbWikiRow, index: number) => void;
+}
+
+const AddWikiItemButton: React.FC<AddWikiItemButtonProps> = ({
+    doesEmptyRowExist,
+    appendNewRow,
+}) => {
+    const addWikiItem = async (): Promise<void> => {
+        if (!doesEmptyRowExist) {
+            const { data, error } = (await supabase
+                .from("wiki")
+                .insert({})
+                .select()
+                .single()) as WikiRowQueryType;
+            error
+                ? logErrorReturnLogId("error inserting and fetching new data", error)
+                : appendNewRow(data, -1);
+        }
```

Comment:
This is a very nit-picky, stylistic comment, but can we use the early-return pattern here? The idea is to do validation checks at the start of a function and return if they aren't met. This allows you to nest as little code inside if statements as possible, and to keep the main purpose of the function at the root, handling the edge cases in `if`s. In practice, that'll look like this:
```
    const addWikiItem = async (): Promise<void> => {
        if (doesEmptyRowExist) {
            return;
        }
        const { data, error } = (await supabase
            .from("wiki")
            .insert({})
            .select()
            .single()) as WikiRowQueryType;
        error
            ? logErrorReturnLogId("error inserting and fetching new data", error)
            : appendNewRow(data, -1);
    };
```